pilot/pkg/dns/proxy.go

```diff
@@ -46,9 +46,9 @@ func newDNSProxy(protocol string, resolver *LocalDNSServer) (*dnsProxy, error) {
 	p.downstreamMux.Handle(".", p)
 	p.downstreamServer.Handler = p.downstreamMux
 	if protocol == "udp" {
-		p.downstreamServer.PacketConn, err = net.ListenPacket("udp", ":15053")
+		p.downstreamServer.PacketConn, err = net.ListenPacket("udp", "127.0.0.1:15053")
```

Comment:
I think we will want `localhost` instead to support ipv6?

Reply:
Thanks a lot for your comment. I agree that `localhost` should do better.

However, please kindly note that `net.Listen()` on a host name is [not recommended](https://github.com/golang/go/blob/master/src/net/dial.go#L694) and would [only pick the first IPv4 address](https://github.com/golang/go/issues/9334). I.e., this may eventually fall back to `127.0.0.1` in most of the cases. And comparing w/ the original intention of listening at `:15053`, listening on `localhost` might not work well on IPv6-only cases (e.g. the similar [cilium case](https://github.com/cilium/cilium/issues/13165)).

A better solution could be handling separately for IPv4 and IPv6 using literal IP addresses on the new DNS proxy init, by leveraging the `IsIPv6` agent config. But I guess this could be addressed via another thread if considered to be more appropriate.

Reply:
BTW, it seems to me that istio-iptables does not handle IPv6 completely for the DNS redirection case. Shall we take care of that too? Thanks!